cli/src/build.rs

```diff
@@ -86,135 +70,108 @@ pub fn build_project(
+        copy_mopro_wasm_lib()?;
+    }
+
+    // Notification when the user selects the 'halo2' adapter and includes the 'web' platform in the selection.
+    if config.adpater_contains(Adapter::Halo2) && platform.contains(Platform::Web) {
+        let confirm = Confirm::with_theme(&ColorfulTheme::default())
                 .with_prompt("Halo2 WASM code will only be generated for the web platform. Do you want to continue?")
                 .default(true)
                 .interact()?;
 
-            if !confirm {
-                style::print_yellow("Aborted build for web platform".to_string());
-                std::process::exit(0);
-            }
-
-            copy_mopro_wasm_lib()?;
+        if !confirm {
+            style::print_yellow("Aborted build for web platform".to_string());
+            std::process::exit(0);
         }
 
-        // Archtecture selection for iOS or Andriod
-        let mut selected_architectures: HashMap<String, Vec<String>> = HashMap::new();
-
-        for platform in &selected_platforms {
-            let archs = match platform.as_str() {
-                "ios" => select_architectures("iOS", &IOS_ARCHS)?,
-                "android" => select_architectures("Android", &ANDROID_ARCHS)?,
-                _ => vec![],
-            };
-
-            selected_architectures.insert(platform.clone(), archs);
-        }
-
-        for platform in selected_platforms.clone() {
-            let arch_key: &str = match platform.as_str() {
-                "ios" => "IOS_ARCHS",
-                "android" => "ANDROID_ARCHS",
-                "web" => "",
-                _ => unreachable!(),
-            };
-
-            let selected_arch = selected_architectures
-                .get(&platform)
-                .map(|archs| archs.join(","))
-                .unwrap_or_default();
-
-            let status = std::process::Command::new("cargo")
-                .arg("run")
-                .arg("--bin")
-                .arg(platform.clone())
-                .env("CONFIGURATION", mode.clone())
-                .env(arch_key, selected_arch)
-                .status()?;
+        copy_mopro_wasm_lib()?;
+    }
 
-            // Add only successfully compiled platforms to the config.
-            if status.success() {
-                config.target_platforms.insert(platform.to_string());
-            } else {
-                // Return a custom error if the command fails
-                return Err(anyhow::anyhow!(
-                    "Output with status code {}",
-                    status.code().unwrap()
-                ));
-            }
+    // Archtecture selection for iOS or Andriod
+    let selected_architectures = platform.select_archs();
+
+    for p in platform.platforms.clone() {
+        let platform_str: &str = p.into();
+        let selected_arch = selected_architectures
+            .get(platform_str)
+            .map(|archs| archs.join(","))
+            .unwrap_or_default();
```

Comment:
I found that 
`selected_architectures` is `{"iOS": ["aarch64-apple-ios"]}`
but `platform_str` is `"ios"`
so it will select a `None`